packages/sdk/src/oracle/services/statemachines/pollNewEvents.ts

```diff
@@ -46,6 +46,9 @@ export function Handlers(store: Store): GenericHandlers<Params, Memory> {
           memory.lastBlock = latestBlock;
           // just count how many successful iterations we do as a kind of sanity check
           memory.iterations++;
+        } else {
+          // if we dont have a lastblock set, set it to our current block
+          memory.lastBlock = memory.lastBlock || currentBlock;
```

Comment:
this was a bug, since memory.lastblock was never initialized, it was always "currentBlock" which mean the logic above this never ran. so now if its not set, set it to the current block and it will run next iteration